utils/wallet.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Hash,
+  PublicClient,
+  SimulateContractParameters,
+  WriteContractParameters,
+} from 'viem'
+import { GetWalletClientResult } from '@wagmi/core'
+
+export async function writeContract(
+  publicClient: PublicClient,
+  walletClient: GetWalletClientResult,
+  args: WriteContractParameters | SimulateContractParameters,
+): Promise<Hash | undefined> {
+  if (!walletClient) {
+    return
+  }
+  const useSimulator =
```

Comment:
`useSimulate` sounds more natural than `useSimulator`.
`useSimulate` should default to true as that's what the production build will use.
The boolean can be simplified like so. 
```
const useSimulate = !(process.env.NEXT_PUBLIC_USE_SIMULATE === 'false')
```

Reply:
https://github.com/clober-dex/coupon.finance/pull/110/commits/66ac336e9ff5f9b7d6d72027a802593c00a8cf70